core/src/main/java/nu/yona/server/rest/ErrorLoggingFilter.java

```diff
@@ -0,0 +1,63 @@
+package nu.yona.server.rest;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatus.Series;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ErrorLoggingFilter implements Filter
+{
+	private static final Logger logger = LoggerFactory.getLogger(ErrorLoggingFilter.class);
+
+	@Override
+	public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain chain)
+			throws IOException, ServletException
+	{
+		HttpServletRequest request = (HttpServletRequest) servletRequest;
+		HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+		chain.doFilter(request, response);
+
+		if (HttpStatus.Series.valueOf(response.getStatus()) == Series.SUCCESSFUL)
```

Comment:
Are 500 errors not logged already because they are related to exceptions? So you could also filter for the 400 series (invalid request). Although logging errors two times is not very problematic.